### PR TITLE
chore: fix some missing php-doc

### PR DESCRIPTION
**Pull request by yassinehamouten**
Some missing return and thorw in php doc

### CODE REVIEW

**Comment by ismail1432** [comment]
thanks for your contribution, IMO adding phpdoc with class name and @return don't help very much and add some noise.
Having return type in function is enough

File: src/AirtableClient.php [hunk 1]
```diff
@@ -5,11 +5,12 @@
 namespace Yoanbernabeu\AirtableClientBundle;
 
 use Symfony\Component\Serializer\Normalizer\ObjectNormalizer;
+use Symfony\Contracts\HttpClient\Exception\TransportExceptionInterface;
 use Symfony\Contracts\HttpClient\HttpClientInterface;
 use Symfony\Contracts\HttpClient\ResponseInterface;
 
 /**
- * AirtableClient.
+ * Class AirtableClient.
```
Comment (ismail1432) [inline src/AirtableClient.php]: You can remove AirtableClient here

File: src/AirtableClient.php [hunk 2]
```diff
@@ -113,6 +114,12 @@ public function findTheLatest(string $table, $field, ?string $dataClass = null):
 
     /**
      * Use the HttpClient to request Airtable API and returns the response.
+     *
+     * @param string $url
+     *
+     * @throws TransportExceptionInterface
+     *
+     * @return ResponseInterface
```
Comment (ismail1432) [inline src/AirtableClient.php]: Can be remove

File: src/AirtableClient.php [hunk 3]
```diff
@@ -113,6 +114,12 @@ public function findTheLatest(string $table, $field, ?string $dataClass = null):
 
     /**
      * Use the HttpClient to request Airtable API and returns the response.
+     *
+     * @param string $url
```
Comment (ismail1432) [inline src/AirtableClient.php]: To remove

File: src/AirtableClientBundle.php [hunk 4]
```diff
@@ -4,15 +4,21 @@
 
 namespace Yoanbernabeu\AirtableClientBundle;
 
+use Symfony\Component\DependencyInjection\Extension\ExtensionInterface;
 use Symfony\Component\HttpKernel\Bundle\Bundle;
 use Yoanbernabeu\AirtableClientBundle\DependencyInjection\AirtableClientExtension;
 
+/**
+ * Class AirtableClientBundle.
```
Comment (ismail1432) [inline src/AirtableClientBundle.php]: Remove

File: src/AirtableClientBundle.php [hunk 5]
```diff
@@ -4,15 +4,21 @@
...
+ */
 class AirtableClientBundle extends Bundle
 {
     /**
      * Overridden to allow for the custom extension alias.
+     *
+     * @return ExtensionInterface
```
Comment (ismail1432) [inline src/AirtableClientBundle.php]: Remove

File: src/AirtableClientInterface.php [hunk 6]
```diff
@@ -4,6 +4,9 @@
 
 namespace Yoanbernabeu\AirtableClientBundle;
 
+/**
+ * Interface AirtableClientInterface.
```
Comment (ismail1432) [inline src/AirtableClientInterface.php]: This phpdoc don't add benefit, can you remove please?

File: src/AirtableClientInterface.php [hunk 7]
```diff
@@ -35,6 +38,8 @@ public function findBy(string $table, string $field, string $value, ?string $dat
      * @param string      $table     Table Name
      * @param string      $id        Id
      * @param string|null $dataClass The name of the class which will hold fields data
+     *
+     * @return AirtableRecord|null
```
Comment (ismail1432) [inline src/AirtableClientInterface.php]: Same here

Comment (ismail1432) [inline src/AirtableClientInterface.php]: Same here

**Review by ismail1432 — COMMENTED** [review]
thanks for your contribution, IMO adding phpdoc with class name and @return don't help very much and add some noise.
Having return type in function is enough
My 2 cents

File: .php-cs-fixer.dist.php [hunk 8]
```diff
@@ -34,6 +34,7 @@
         'combine_consecutive_unsets' => true,
         'compact_nullable_typehint' => true,
         'no_superfluous_elseif' => true,
+        'no_superfluous_phpdoc_tags' => false,
```
Comment (ismail1432) [inline .php-cs-fixer.dist.php]: 😢

File: src/AirtableClient.php [hunk 9]
```diff
@@ -129,7 +136,7 @@ private function request(string $url): ResponseInterface
      * Turns an array of arrays to an array of AirtableRecord objects.
      *
      * @param array  $records   An array of arrays
-     * @param string $dataClass Optionnal class name which will hold record's fields
+     * @param string $dataClass Optional class name which will hold record's fields
```
Comment (ismail1432) [inline src/AirtableClient.php:130]: 👍🏽

File: src/AirtableRecord.php [hunk 10]
```diff
@@ -10,22 +10,27 @@
...
+/**
+ * Class AirtableRecord.
+ */
 final class AirtableRecord
 {
-    /**
-     * @var object|array<array-key, mixed>
-     */
-    private $fields;
+    /** @var string */
```
Comment (ismail1432) [inline src/AirtableRecord.php]: Remove here and other

**Review by liorchamla — COMMENTED** [review]
I am a bit concerned by those changes and the comments from @ismail1432 as I know that with modern PHP versions, TypeHinting seems to be good documentation, but I kinda like the fact that we also find the same concepts inside PHPDoc.

Problem : when we move some code, we also have to edit PHPDocs, so I can understand why people sometimes just wanna get rid of the PHPDocs.

Let @yoanbernabeu decide what will be his take on this :)

**Comment by yoanbernabeu** [comment]
Thanks for the suggestions.

I too am hesitant.
I like the idea that the PHPdoc is very complete, but actually, with TypeHitting, the code is self-documenting?

I am in favor of limiting the use of PHPdoc to what is strictly necessary.
What do you think ?

**Comment by yassinehamouten** [comment]
Hello, 
Sorry i was busy these last few days.

I prefere to have the phpdoc + type hinting. But I understand your opinions.

So I will revert changes about php-doc 👍

**Comment by yoanbernabeu** [comment]
I propose to close this PR.
Thanks for the proposal.

File: src/AirtableClient.php (hunk 3 above)
Comment (yassinehamouten) [reply src/AirtableClient.php]: Why we have to remove this ?

**Comment by yassinehamouten** [comment]
Hello, 
Sorry I was on vacation. 
This PR is now ok I think

**Pull request closed without merging**